Documentation/Input/CreateDataProvider.md

```diff
@@ -0,0 +1,202 @@
+
+For digital (button) type controls, raise the OnInputDown and OnInputUp events.
+
+``` c#
+// inputAction is the input event that is to be raised.
+
+if (interactionSourceState.touchpadPressed)
+{
+    InputSystem?.RaiseOnInputDown(InputSource, ControllerHandedness, inputAction);
+}
+else
+{
+    InputSystem?.RaiseOnInputUp(InputSource, ControllerHandedness, inputAction);
+}
+```
+
+For analog controls (ex: touchpad position) the InputChanged event should be raised.
+
+``` c#
+InputSystem?.RaisePositionInputChanged(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction, interactionSourceState.touchpadPosition);
+```
+
+### Apply the MixedRealityDataProvider attribute
+
+The final step of creating an input system data provider is to apply the [`MixedRealityDataProvider`](xref:Microsoft.MixedReality.Toolkit.MixedRealityDataProviderAttribute)
+attribute to the class. This step enables setting the default profile and platform(s) for the provider, when selected in the input system profile.
+
+``` c#
+[MixedRealityDataProvider(
+    typeof(IMixedRealityInputSystem),
+    SupportedPlatforms.WindowsUniversal,
+    "Windows Mixed Reality Device Manager")]
+public class WindowsMixedRealityDeviceManager : BaseInputDeviceManager, IMixedRealityCapabilityCheck
+{ }
+```
+
+## Create the profile and inspector
+
+In the Mixed Reality Toolkit, data providers are configured using [profiles](../Profiles/Profiles.md).
+
+Data providers with additional configuration options (ex: [InputSimulationService](../InputSimulation/InputSimulationService.md)) should create a profile and inspector to allow
+customers to modify the behavior to best suit the needs of the application.
+
+> The complete code for the example in this section can be found in the MixedRealityToolkit.Services\InputSimulation folder.
+
+### Define the profile
+
+Profile contents should mirror the accessible properties of the observer (ex: update interval). All of the user configurable properties defined in each
+interface should be contained with the profile.
+
+``` c#
+[CreateAssetMenu(
+    menuName = "Mixed Reality Toolkit/Profiles/Mixed Reality Simulated Input Profile",
+    fileName = "MixedRealityInputSimulationProfile",
+    order = (int)CreateProfileMenuItemIndices.InputSimulation)]
+public class MixedRealityInputSimulationProfile : BaseMixedRealityProfile
+{ }
+```
+
+The `CreateAssetMenu` attribute can be applied to the profile class to enable customers to create a profile instance using the 
+**Create > Assets > Mixed Reality Toolkit > Profiles** menu.
+
+### Implement the inspector
+
+Profile inspectors are the user interface for configuring and viewing profile contents. Each profile inspector should extend the
+[`BaseMixedRealityToolkitConfigurationProfileInspector]() class.
+
+``` c#
+[CustomEditor(typeof(MixedRealityInputSimulationProfile))]
+public class SpatialObjectMeshObserverProfileInspector : BaseMixedRealityToolkitConfigurationProfileInspector
```

Comment:
>SpatialObjectMeshObserverProfileInspector [](start = 13, length = 41)

I think that this should still be related to the input simulation profile, not the spatial observer

Otherwise this part doesn't make sense to me:

[CustomEditor(typeof(MixedRealityInputSimulationProfile))]
public class SpatialObjectMeshObserverProfileInspector

Reply:
@keveleigh pointed this out and submitted a suggested fix. it was a copy-paste error on my part

:)